aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -941,6 +947,72 @@ public void run() {
         }, "Mqtt Connect Thread").start();
     }
 
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using custom authorization tokens.
+     *
+     * @param tokenKeyName This specifies the key name that your device chooses, which indicates the
+     *                     token in the custom authorization HTTP request header. It is a string;
+     *                     the minimum length is 1 character, and the maximum length is 128 characters.
+     * @param token This specifies the custom authorization token to authorize the request to the AWS
+     *              IoT gateway. It is a string; the minimum length is 1 character, and the maximum
+     *              length is 1,024 characters.
+     * @param tokenSignature This is the token signature generated by using the private key with the
+     *                       SHA256withRSA algorithm.
+     * @param customAuthorizer Name of the AWS IoT custom authorizer.
+     * @param statusCallback When a new MQTT session status is received, a function of callback will
+     *                       be called with new connection status.
+     */
+    public void connect(
+            final String tokenKeyName,
+            final String token,
+            final String tokenSignature,
+            final String customAuthorizer,
+            final AWSIotMqttClientStatusCallback statusCallback) {
+
+        if (tokenKeyName == null || token == null || tokenSignature == null) {
+            throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
+        }
+
+        this.userStatusCallback = statusCallback;
+
+        // Do nothing if Connecting, Connected or Reconnecting
+        if (connectionState != MqttManagerConnectionState.Disconnected) {
+            userConnectionCallback();
+            return;
+        }
+
+        authMode = AuthenticationMode.CUSTOM_AUTH;
+        String endpointWithHttpPort = getEndpointWithHttpPort();
+        LOGGER.debug("MQTT broker endpoint: " + endpointWithHttpPort);
+        try {
+            if (mqttClient == null) {
+                mqttClient = new MqttAsyncClient(
+                        "wss://" + endpointWithHttpPort + "/mqtt",
+                        mqttClientId,
+                        new MemoryPersistence());
+            }
+
+            customWebsocketHeaders = new Properties();
```

Comment:
Where is this declared? Can it be local to this method? Or alternately, can you instantiate the `new Properties()` closer to the declaration (Constructor?)

Reply:
This is an instance variable because it is reused for reconnection. I have moved the instantiation to Constructor anyways.